### ISSUE TITLE
Add --reserve-disk option to service create

### ISSUE THREAD

**Issue by shane-axiom** [docker/docker#27555]
Current `docker swarm create` has options for `--reserve-cpu` and `--reserve-memory` but no `--reserve-disk`. There should be a way to reserve an amount of disk space for each container.

Issue spawned from discussion on https://github.com/docker/docker/issues/26259#issuecomment-248059992.

**Comment by guenhter** [comment]
Maybe this can also be considered, when this feature is implemented:
https://forums.docker.com/t/i-o-should-also-be-reservable-for-swarm-service/23208

**Comment by thaJeztah** [comment]
ping @dongluochen @aluzzardi

**Comment by johnjelinek** [comment]
any updates on this?

**Comment by thaJeztah** [comment]
No, I haven't seen a lot of interest for this feature, and to my knowledge, no-one has been working on this. A more in-depth design is likely needed (what is considered "disk space for the container"? Does this include the base image's size? the writable layer? volumes?)

**Comment by johnjelinek** [comment]
I was thinking something more simplistic, that if the host does not have
the requested reserve-disk size available at the time of
create/scale/update, then it would go spin up on another host.

**Comment by thaJeztah** [comment]
Oh, yes, I was reading to much into it; this request is only for _reserve_, not _limit_
